File: grupoDeSlides_01-02/ex-09/ex-09_c++/ex-09_intermediaria/ex-09_makefile/Carro.cpp
#include "Carro.hpp"

Carro::Carro(const double &consumo, const double &capacidade) : consumo(consumo),
                                                                capacidade(capacidade),
                                                                combustivel(0)
{
}

Carro::~Carro()
{
}

void Carro::abastecer(const double &quantidade)
{
  // Verifica que não sobrepase a capacidade de combustível do carro.
  if (this->combustivel + quantidade <= this->capacidade)
    this->combustivel += quantidade;
  else
    this->combustivel += this->capacidade - this->combustivel;
}

void Carro::andar(const double &distancia)
{
  // Resta o combustível consumido para andar a distância solicitada.
  this->combustivel -= distancia / this->consumo; // km / (km/litro) = litros consumidos
}

const double &Carro::getCombustivel() const
{
  return this->combustivel;
}

File: grupoDeSlides_01-02/ex-07/ex-07_java/ex-07/src/ex7carro/Carro.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ex7carro;

/**
 *
 * @author Robson
 */
public class Carro {
    
    private double consumo; //km/litro
    private double capacidade; //litro
    private double qtde;       //litro
    
    //construtora
    public Carro (double cons, double cap)
    {
        consumo = cons;
        capacidade = cap;
        //tanque inicialmente vazio
        qtde = 0;
    }
    public double andar (double dist)
    {
        
        if (qtde >= dist/consumo)
        {
        
            //cuidado! Quanto maior o valor
            //de "consumo", mais econômico
            //é o veículo!
            qtde = qtde - dist/consumo;
            return dist;
        }
        else
        {
            //se não tem combustivel
            //suficiente, anda o que é
            //possível e esvazia o 
            //tanque.
            double aux = qtde*consumo;
            qtde = 0;
            return aux;
        }
    }
    public double getCombustivel() { return qtde; }
    public double abastecer(double q)
    {
        if (q > capacidade - qtde)
        {
            double aux = capacidade - qtde;
            qtde = capacidade;
            return aux;
        }
        else
        {
            qtde += q;
            return q;
        }
    
    
    }
    
    
}
